pointcloud/basic_octree.go

```diff
@@ -198,22 +202,21 @@ func (octree *BasicOctree) ToProtobuf() *commonpb.Geometry {
 	return nil
 }
 
-// CollidesWithGeometry will return whether a given geometry is in collision with a given point.
-// A point is in collision if its stored probability is >= confidenceThreshold and if it is at most buffer distance away.
-func (octree *BasicOctree) CollidesWithGeometry(
-	geom spatialmath.Geometry,
-	confidenceThreshold int,
-	buffer,
-	collisionBufferMM float64,
-) (bool, error) {
-	if octree.MaxVal() < confidenceThreshold {
+// CollidesWith checks if the given octree collides with the given geometry and returns true if it does.
+// A point is in collision if its stored probability is >= confidenceThreshold and if it is at most collisionBufferMM distance away.
+func (octree *BasicOctree) CollidesWith(geom spatialmath.Geometry, collisionBufferMM float64) (bool, error) {
+	if octree.MaxVal() < octree.confidenceThreshold {
```

Comment:
It's entirely possible for octree node values to wind up with confidence values well below zero; indeed this ticket was inspired by finding an octree whose confidence values were the min int64.

Reply:
And you can also prefilter and mutate as needed.  This isn't intended to be a catch all for every case.  I just want to preserve the behavior of SLAM based planning as is

Reply:
Well, no, this is actually worse than what currently exists.
I would naively expect something with a comment saying "this is a value between 0-100" to then provide values in that range.
Furthermore, if I set "0" as the min value, I would expect checks to simply be skipped. Instead, if I set "0" in my "0-100" range, and I wind up having a pcd with values below zero, then I filter everything.
And on top of that, with the current value check, if I were to pass a below-zero value because I *knew* I was receiving a pcd with below-zero values, it would get updated to the default of 50. With this change there is no way to actually check collisions against a pcd with below-zero values/

Reply:
Hey the ticket I’m trying to resolve is unrelated entirely to confidence. And I’m not going to break slam based motion as a side effect here by removing it entirely.  You indicated previously that allowing this to be configurable would be acceptable.  I did this but now you are claiming this is worse than what exists.  I am unclear how to proceed based on these conflicting comments.

Reply:
Sorry- didn't mean to be contradictory. I agree, slam motion should not be broken.

The root issue here is that we've created a general-purpose data structure, and then baked in application-specific assumptions, which are now biting us. If I recall, the `CollidesWithGeometry` method was the last attempt to work around the problem. The work here does a lot more to get the data structure to a better, more general place for sure.

I think with the removal of the restriction and default setting on the confidence range, this is fine with me.

What I *wish* for is as way to obviously say "ignore confidence" without knowing anything about the actual values present in the pcd, but I'm not sure exactly what that would look like at the moment, so I think this is good.

Reply:
Ok after syncing offline we have agreed that setting thr `emptyProb` variable to zero would be satisfactory to our goals.  Users can then make the confidence threshold <=0 to disable confidence based checking